modules/images/webp-uploads/rest-api.php

```diff
@@ -23,20 +23,30 @@ function webp_uploads_update_rest_attachment( WP_REST_Response $response, WP_Pos
 	}
 
 	$metadata = wp_get_attachment_metadata( $post->ID );
-	foreach ( $data['media_details']['sizes'] as $size => $details ) {
+	foreach ( $data['media_details']['sizes'] as $size => &$details ) {
 		if ( empty( $metadata['sizes'][ $size ]['sources'] ) || ! is_array( $metadata['sizes'][ $size ]['sources'] ) ) {
 			continue;
 		}
-
 		$sources   = array();
-		$directory = dirname( $data['media_details']['sizes'][ $size ]['source_url'] );
+		$directory = dirname( $details['source_url'] );
 		foreach ( $metadata['sizes'][ $size ]['sources'] as $mime => $mime_details ) {
+			$source_url                  = "{$directory}/{$mime_details['file']}";
+			$mime_details['source_url']  = $source_url;
+			$details['sources'][ $mime ] = $mime_details;
+		}
+	}
+
+	$full_src = wp_get_attachment_image_src( $post->ID, 'full' );
+
+	if ( ! empty( $full_src ) ) {
```

Comment:
We should also check here whether all other necessary data is set, just to be safe:
```suggestion
	if ( ! empty( $full_src ) && ! empty( $data['media_details']['sources'] ) && ! empty( $data['media_details']['sizes']['full'] ) ) {
```
If any of this is not here for some reason, we also can't / shouldn't set it.

Reply:
Make sense 👍